src/main/java/hudson/plugins/s3/Uploads.java

```diff
@@ -21,7 +21,7 @@ private Uploads() {}
     private final transient HashMap<FilePath, Upload> startedUploads = new HashMap<>();
     private final transient HashMap<FilePath, InputStream> openedStreams = new HashMap<>();
 
-    public void startUploading(TransferManager manager, FilePath file, InputStream inputsStream, String bucketName, String objectName, ObjectMetadata metadata) throws AmazonServiceException {
+    public Upload startUploading(TransferManager manager, FilePath file, InputStream inputsStream, String bucketName, String objectName, ObjectMetadata metadata) throws AmazonServiceException {
```

Comment:
It is, in general, a bad idea to separate creation of a handle and looking it up. It's a source of race conditions, if something happens between the creation and lookup and the old handle disappears or is displaced. In this case this is perhaps not a problem, but I still consider it bad style. Returning the value directly upon creation is much safer and cleaner.

Changing a method to return a value instead of `void` is in general quite safe, since there cannot be any previous users of the return value.

I can still do it the way you suggest, but please confirm that you really insist on it.